features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -506,6 +506,43 @@ ble_error_t GenericSecurityManager::requestAuthentication(connection_handle_t co
 // MITM
 //
 
+ble_error_t GenericSecurityManager::generateOOB(
+    const address_t *address
+) {
+    /* legacy pairing */
+    ble_error_t status = get_random_data(_oob_temporary_key.buffer(), 16);
+
+    if (status == BLE_ERROR_NONE) {
+        _oob_temporary_key_creator_address = *address;
+
+        eventHandler->legacyPairingOobGenerated(
+            &_oob_temporary_key_creator_address,
+            &_oob_temporary_key
+        );
+    } else {
+        return status;
+    }
+
+    /* Secure connections. Avoid generating if we're already waiting for it.
+     * If a local random is set to 0 it means we're already calculating
+     * unless the address is invalid which means we've never done so yet */
+    if (_oob_local_random != 0 || _oob_local_address == address_t()) {
```

Comment:
I do not see where `_oob_local_address` is reset to the default value.
We should be able to generate oob data more than once during the application lifetime.

Reply:
address is reset by the constructor, then random is set when generation is complete so we can recalculate then

Reply:
actually I can make it simpler by simply setting the random to a nonzero value in the constructor which will avoid having to rely on an invalid address for the first generation